### PR TITLE
Basic Samurai Support added

### PR DESCRIPTION
**Pull request by Rjb99**
Added sam actions and statuses

used the base cast times for Iaijutsu skills.

### CODE REVIEW

File: src/data/ACTIONS/SAM.js [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+		name: 'Hagakure'
+		icon: 'https://secure.xivdb.com/img/game/003000/003176.png'
+		onGCD: false,
+		cooldown: 40,
+	},
+
+	HISSATSU_GUREN:
+	{
+		id:7496,
+		name: 'Hisstasu: Guren',
```
Comment (NamelessFaceless) [inline src/data/ACTIONS/SAM.js]: just a typo. xd

File: src/data/ACTIONS/SAM.js [hunk 2]
```diff
@@ -0,0 +1,240 @@
+// Samurai Actions
+export default {
+	//-----
+	//Player GCDs
+	//-----
+	HAKAZE:
```
Comment (ackwell) [inline src/data/ACTIONS/SAM.js]: Not sure if `eslint` picks this up, but please follow the convention in the rest of the codebase for this, e.g.:

```js
export default {
	HAKAZE: { // on same line
		key: value, // space between key and val
	},
}
```

File: src/data/ACTIONS/SAM.js (hunk 1 above)
Comment (ackwell) [reply src/data/ACTIONS/SAM.js]: ☝️

File: src/data/STATUSES/SAM.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
+export default {
+	
+	THIRD_EYE:
```
Comment (ackwell) [inline src/data/STATUSES/SAM.js]: Same deal as actions file

File: src/data/ACTIONS/SAM.js [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+export default {
+	//-----
+	//Player GCDs
+	//-----
+	HAKAZE: 
+	{
+        	id:7477,
+        	name: 'Hakazae',
+        	icon: 'https://secure.xivdb.com/img/game/003000/003151.png',
+        	onGCD: true,
```
Comment (ackwell) [inline src/data/ACTIONS/SAM.js]: `onGcd`

File: src/data/ACTIONS/SAM.js [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+	//-----
+	//Player OGCDs
+	//-----
+	
+	THIRD_EYE:
+	{
+		id:7498,
+		name:'Third Eye',
+		icon:'https://secure.xivdb.com/img/game/003000/003153.png'
+		onGCD: false,
```
Comment (ackwell) [inline src/data/ACTIONS/SAM.js]: No need to include `onGcd` when it isn't one

File: src/parser/jobs/sam/About.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+		<p>This analyzer aims to identify some of the low-hanging fruit that could be used to improve your SAM gameplay, as well as give a deeper insight into what happened during an encounter.</p>
+		<Message warning icon>
+			<Icon name="warning sign"/>
+			<Message.Content>
+				<b>The module is still throughly incomplete, this is only the most barebones support for <em>basic</em> analysis of SAM gameplay.</b> And while the existing features below should be reasonably accurate, this system <em>is</em> still in development, and may get a little mixed up sometimes. If you notice any issues, or have any concerns, please drop by our Discord channel or report a bug on our github repository!
+			</Message.Content>
+		</Message>
+	</Fragment>
+	supportedPatch = '4.35'
+	contributors =
```
Comment (ackwell) [inline src/parser/jobs/sam/About.js]: Looks like something happened w/ indentation here.

File: src/parser/jobs/sam/About.js [hunk 7]
```diff
@@ -0,0 +1,23 @@
+import React, {Fragment} from 'react'
+import {Icon, Message} from 'semantic-ui-react'
+
+
+import CoreAbout from 'parser/core/modules/About'
+
+export default class About extends CoreAbout {
+	description = <Fragment>
```
Comment (ackwell) [inline src/parser/jobs/sam/About.js]: I won't block on this, but see the pins in `#development` re: descriptions.

File: src/parser/jobs/sam/Higanbana.js [hunk 8]
```diff
@@ -0,0 +1,105 @@
+import React, {Fragment} from 'react'
+
+import {ActionLink} from 'components/ui/DbLink'
+import ACTIONS from 'data/ACTIONS'
+import STATUSES from 'data/STATUSES'
+import Module from 'parser/core/Module'
+import {Rule, Requirement} from 'parser/core/modules/Checklist'
+import {Suggestion, SEVERITY} from 'parser/core/modules/Suggestions'
+
+const STATUS_DURATION =
```
Comment (ackwell) [inline src/parser/jobs/sam/Higanbana.js]: Indentation

**Comment by Rjb99** [comment]
changed the code to match the style changes requested and fixed all the onGCDs

added bare bones samurai support:
-higanbana up time is tracked
-kaiten usage on GCDs that are not sen moves is also tracked

also updated several wrong ids (whoops) and added statuses for the dot and kaiten buff.

**Pull request merged**